### ISSUE TITLE
Wrong implementation of DANN

### ISSUE THREAD

**Issue by lzhangbj** [virajprabhu/CLUE#4]
Hi, I realized that you have used the wrong implementation of DANN(revgrad). Specifically, you attach the discriminator after the classifier instead of the penultimate layer. see here:
https://github.com/virajprabhu/CLUE/blob/99cb2b498fe42ab8ba8425eebb021e8fba7ec960/adapt/models/adapt_net.py#L49
But in original paper, and in every other paper's reimplementation, they follow the illustration below:

<img width="713" alt="Screen Shot 2022-08-25 at 10 39 18 AM" src="https://user-images.githubusercontent.com/22856801/186695132-bd8d141e-c3d2-4b80-a140-b835003ba09e.png">

Could you explain why you did this?

**Comment by virajprabhu** [comment]
Thanks for pointing this out. My understanding is that either choice is valid for eg. https://github.com/jhoffman/cycada_release/blob/master/cycada/models/adda_net.py#L33 also attaches the discriminator after the logits. Empirically, I have found both to work similarly.
